### PR TITLE
Add file_name argument to load_state_dict_from_url

### PR DESCRIPTION
**Pull request by MohamedAliRashad**
Add the feature proposed here #39196

### CODE REVIEW

**Comment by MohamedAliRashad** [comment]
@ailzhang I hope this one is better

**Review by ailzhang — APPROVED** [review]
Thanks @MohamedAliRashad ! I just added a small test to make sure we don't regress :D

File: torch/hub.py [hunk 1]
```diff
@@ -423,7 +423,7 @@ def _download_url_to_file(url, dst, hash_prefix=None, progress=True):
             _download_url_to_file will be removed in after 1.3 release')
     download_url_to_file(url, dst, hash_prefix, progress)
 
-def load_state_dict_from_url(url, model_dir=None, map_location=None, progress=True, check_hash=False):
+def load_state_dict_from_url(url, model_dir=None, file_name=None, map_location=None, progress=True, check_hash=False):
```
Comment (SsnL) [inline torch/hub.py:426]: adding an arg in the middle is BC breaking

Comment (MohamedAliRashad) [reply torch/hub.py:426]: I found it more sensible to add after `model_dir`, i am okay with changing the position if it will maintain compatibility though

**Pull request closed without merging**